subsys/bluetooth/mesh/settings.c

```diff
@@ -117,7 +117,7 @@ static int mesh_commit(void)
 	return 0;
 }
 
-SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh, "bt/mesh", NULL, NULL, mesh_commit,
+SETTINGS_STATIC_HANDLER_DEFINE(_2_bt_mesh, "bt/mesh", NULL, NULL, mesh_commit,
```

Comment:
Shouldn't `BT_MESH_SETTINGS_DEFINE` use this prefix as well?

Reply:
```
#define BT_MESH_SETTINGS_DEFINE(_hname, _subtree, _set)
```
 only define set, not commit


``` C
int settings_load_subtree(const char *subtree)
{
	struct settings_store *cs;
	int rc;
	const struct settings_load_arg arg = {
		.subtree = subtree
	};

	/*
	 * for every config store
	 *    load config
	 *    apply config
	 *    commit all
	 */
	k_mutex_lock(&settings_lock, K_FOREVER);
	SYS_SLIST_FOR_EACH_CONTAINER(&settings_load_srcs, cs, cs_next) {
		cs->cs_itf->csi_load(cs, &arg);
	}
	rc = settings_commit_subtree(subtree);
	k_mutex_unlock(&settings_lock);
	return rc;
}
```

The most important thing is the order of commits

Reply:
> There are few more places where SETTINGS_STATIC_HANDLER_DEFINE is used by bluetooth related models. Shouldn't they be updated as well?
controller/ll_sw/ll_settings.c:92:SETTINGS_STATIC_HANDLER_DEFINE(bt_ctlr, "bt/ctlr", NULL, ctlr_set, NULL, NULL);
host/settings.h:16:	SETTINGS_STATIC_HANDLER_DEFINE(bt_##_hname, "bt/" _subtree, NULL, _set, _commit, NULL)
host/settings.c:284:SETTINGS_STATIC_HANDLER_DEFINE(bt, "bt", NULL, set_setting, commit_settings, NULL);
host/classic/keys_br.c:220:SETTINGS_STATIC_HANDLER_DEFINE(bt_link_key, "bt/link_key", NULL, link_key_set,
mesh/dfu_slot.c:401:SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh_dfu_slots, DFU_SLOT_SETTINGS_PATH, NULL,
mesh/settings.h:39:	SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh_##_hname, "bt/mesh/" _subtree, NULL, pre_##_set,    \
mesh/settings.c:120:SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh, "bt/mesh", NULL, NULL, mesh_commit,
services/dis.c:453:SETTINGS_STATIC_HANDLER_DEFINE(bt_dis, "bt/dis", NULL, dis_set, NULL, NULL);

I confirmed that these are just sets, not commits.

This is also a reply to the third point above. https://github.com/zephyrproject-rtos/zephyr/pull/79690#pullrequestreview-2362040505

Reply:
But if someone adds a commit handler for some of mesh settings, this can cause a bug, can't it?

Reply:
Before this PR, if I did what you said, there would still be a problem of status being out of sync.

Reply:
Mainly inspired by: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/gatt.c#L234